lib/ModelMixins/CatalogMemberMixin.ts

```diff
@@ -180,16 +181,32 @@ function CatalogMemberMixin<T extends Constructor<CatalogMember>>(Base: T) {
             );
             const value = dim.options.find(o => o.id === selectedId)?.value;
             if (isDefined(value)) {
-              updateModelFromJson(this, stratumId, value).raiseError(
+              const result = updateModelFromJson(this, stratumId, value);
+              result.raiseError(
                 this.terria,
                 `Failed to update catalog item ${getName(this)}`
               );
+
+              // If no error and modelDimensionCallLoadMapItems is true, then call loadMapItems
+              if (
+                !result.error &&
+                this.modelDimensionCallLoadMapItems &&
+                MappableMixin.isMixedInto(this)
+              ) {
+                this.loadMapItems().then(loadMapItemsResult => {
+                  loadMapItemsResult.raiseError(this.terria);
+                });
+              }
             }
           }
         })) ?? []
       );
     }
 
+    @computed get viewingControls(): ViewingControl[] {
+      return [];
+    }
+
```

Comment:
I love this addition.

Not part of this PR but one other consumer for viewing controls could be external plugins. External plugins might want to show viewing controls but for specific items or items that match some arbitrary condition, eg, it has a certain mixin.

One way of supporting that use case might be by adding a `Terria` observable called `viewingControlsMenu` which is an array of `(item) => ViewingControl | undefined` and it gets imported here.

Reply:
Yeah I love that idea! that would be super nice